Some layout improvements and optimization for mobile

YuriBrunetto opened the pull request:
I have made some layout improvements, optimization for mobile devices and the app is running with no problems at all. I just have an issue with `stylelint css/*.css` when running `npm test`. Could you please tell me what I did wrong?

# Before & after
You can checkout [this album](http://imgur.com/a/DShf0) to see what I did to improve the layout.

shakeelmohamed commented:
If you click the red X on this page, you'll be taken the travis CI build page that lists the errors from stylelint.
From the screenshots, this change looks good.
I'll give you some better feedback when I have a moment, thanks for doing this!

shakeelmohamed commented on css/styles.css:
```diff
@@ -113,3 +121,90 @@ header, footer {
     background-color: #efefef;
 }
+
+/* improvements by YuriBrunetto */
```
I appreciate you wanting to keep your changes separate, but let's try to introduce some logical ordering. I like that you put the `body` rule above the `header, footer` rule. For example, let's move the `header` and `footer` rules below the `header, footer` rule.

shakeelmohamed commented on css/styles.css:
```diff
@@ -113,3 +121,90 @@ header, footer {
+    margin: 0 auto;
+}
+
+.input {
```
Let's remove this empty rule

shakeelmohamed commented on css/styles.css:
```diff
@@ -113,3 +121,90 @@ header, footer {
+}
+@media all and (min-width:640px) {
+    .sponsor-a {
+        width: 30.666666667%;
```
Is this much decimal precision common practice? I'm not sure :thought_balloon:

YuriBrunetto replied:
It's not very readable, but it's [valid](http://nimbupani.com/using-decimal-percentage-values-in-responsive-design.html). Anyway, if you want we could make this different :)

shakeelmohamed commented on css/styles.css:
```diff
@@ -113,3 +121,90 @@ header, footer {
+    }
+}
+
+.color-666 {
```
This number has some bad mojo around it, let's pick a different one :cloud:

shakeelmohamed commented on css/styles.css:
```diff
@@ -12,16 +16,20 @@ header, footer {
+}
 
-.logo {
+/*.logo {
```
Just remove this rule

shakeelmohamed commented on index.html:
```diff
@@ -30,18 +30,22 @@
-            <a href="https://zenplayer.audio/" title="Zen Audio Player">
-                <img src="img/zen-audio-player-905.png" class="logo" alt="Zen Audio Player logo">
+            <a href="https://zenplayer.audio/" title="Zen Audio Player" class="zen-logo">
+                <img src="img/zen-audio-player-905.png" class="logo" alt="Zen Audio Player logo" width="100">
```
Remove the class attribute, I'd prefer to have a CSS rule for the 100% width

shakeelmohamed commented on index.html:
```diff
@@ -30,18 +30,22 @@
-            <figcaption>Listen to YouTube videos, without the distracting visuals.</figcaption>
         </figure>
+
+        <!-- This should be better on a hgroup -->
```
Cool, I didn't know about this tag. You can remove the comment

shakeelmohamed commented on index.html:
```diff
@@ -92,21 +96,35 @@
-        <p><a href="https://trackjs.com/" target="_blank"><img src="img/trackjs_logo.png" alt="TrackJS Logo"></a></p>
+        <div class="sponsor">
+            <a href="https://browserstack.com" class="sponsor-a" target="_blank">
+                <img src="img/browserstack_logo.png" alt="Browserstack logo" width="100%">
```
as with above, let's make a CSS rule for the 100% width on all of these

shakeelmohamed commented on css/styles.css:
```diff
@@ -113,3 +121,90 @@ header, footer {
+        color: #666 !important;
+    }
+    .btn {
+        height: 50px;
```
This looks vertically stretched too much, if we continue with this height we'll need to increase the width and increase the icon size (possibly via `font-size`).

<img width="504" alt="capturfiles_3" src="https://cloud.githubusercontent.com/assets/1480063/13661154/ef07b44c-e644-11e5-9e6d-c958f3b5fbe8.png">

shakeelmohamed commented on css/styles.css:
```diff
@@ -113,3 +121,90 @@ header, footer {
+    .input-search {
+        height: 50px;
+        font-size: 18px !important;
+        color: #666 !important;
```
Please update all occurrences of `#666` :see_no_evil:

shakeelmohamed commented on css/styles.css:
```diff
@@ -113,3 +121,90 @@ header, footer {
+h2 {
+    font-weight: normal;
+    font-size: 16px;
+    color: #666;
```
Please update all occurrences of `#666` :see_no_evil:, also you can just use multiple classes on the elements that need this color so it's not re-declared so many times

shakeelmohamed commented:
Left several comments on the code, let me know if you need clarification on anything. We're making some progress, I appreciate you taking the initiative to help us rebrand a bit :+1:

YuriBrunetto commented:
Hey, @shakeelmohamed!

I have done every little thing you asked :D 
Can you check if my PR was updated? The new message is `Fix css and html errors in the branch`.

Also, I'm still getting this error (both npm and node are updated here):
![npm error](http://i.imgur.com/X8qHkUi.png)

shakeelmohamed commented:
@YuriBrunetto Great! You're running into a code quality tool we use call stylelint, the build/tests appear to fail because the CSS doesn't pass the stylelint rules we have configured. The rules are configured in the `.stylelintrc` file. I think you can remove the `number-max-precision` rule, but you should rework the rest of the CSS to conform to the rules if possible. 

The things to fix:

```shell
css/styles.css
13:1	Expected empty line before at-rule (at-rule-empty-line-before)
13:27	Expected single space after ":" (media-feature-colon-space-after)
14:5	Expected empty line before nested rule (rule-nested-empty-line-before)
49:1	Expected empty line before at-rule (at-rule-empty-line-before)
49:27	Expected single space after ":" (media-feature-colon-space-after)
50:5	Expected empty line before nested rule (rule-nested-empty-line-before)
158:1	Expected empty line before at-rule (at-rule-empty-line-before)
158:27	Expected single space after ":" (media-feature-colon-space-after)
159:5	Expected empty line before nested rule (rule-nested-empty-line-before)
161:26	Unexpected !important (declaration-no-important)
163:5	Expected empty line before nested rule (rule-nested-empty-line-before)
166:5	Expected empty line before nested rule (rule-nested-empty-line-before)
177:11	Expected single space after ":" (declaration-colon-space-after)
184:1	Expected empty line before at-rule (at-rule-empty-line-before)
184:27	Expected single space after ":" (media-feature-colon-space-after)
185:5	Expected empty line before nested rule (rule-nested-empty-line-before)
186:16	Expected "30.666666667" to be "30.67" (number-max-precision)
191:5	Expected empty line before nested rule (rule-nested-empty-line-before)
197:21	Unexpected !important (declaration-no-important)
202:14	Unexpected unit on zero length number (number-zero-length-no-unit)
```

YuriBrunetto commented:
@shakeelmohamed again, I did everything you asked!

But it's still failing :cry: 
I guess it's because `stylelint` doesn't recognize the `@media` in styles.css

Is this **really** necessary? It's really annoying to keep fixing :l 

Anyway, the changes have been committed agan!

shakeelmohamed commented:
Thanks @YuriBrunetto, I will fix them and send you a PR. Don't worry :+1:

YuriBrunetto commented:
Ok @shakeelmohamed! Just let me know what you did to fix it.

Thanks for being such a nice person and helping me! It's a nice web-app, btw :smile:

shakeelmohamed commented:
@YuriBrunetto Turns out this is just a whitespace issue, I got lazy so I'll let you apply this patch:

```diff
diff --git a/css/styles.css b/css/styles.css
index 92a00b8..ddd244f 100644
--- a/css/styles.css
+++ b/css/styles.css
@@ -11,7 +11,7 @@ header {
     margin: 50px auto 0 auto;
 }
 
-@media all and ( min-width: 480px ) {
+@media all and (min-width: 480px) {
 
     header {
         max-width: 450px;
@@ -50,7 +50,7 @@ h2 {
     margin: 0 auto;
 }
 
-@media all and ( min-width: 480px ) {
+@media all and (min-width: 480px) {
 
     #container {
         max-width: 450px;
@@ -166,7 +166,7 @@ h2 {
     width: 50px;
 }
 
-@media all and ( min-width: 480px ) {
+@media all and (min-width: 480px) {
 
     .input-search {
         height: 50px;
@@ -198,7 +198,7 @@ h2 {
     margin-top: 0;
 }
 
-@media all and ( min-width: 640px ) {
+@media all and (min-width: 640px) {
 
     .sponsor-a {
         width: 30.666666667%;

```

YuriBrunetto commented:
hey @shakeelmohamed! 

turns out it wasn't the whitespace errors

so, I deleted `"media-query-parentheses-space-inside": "always"` and I did some comments in the index.js file. I did that just to pass the build... I guess it's all okay now :smile:

shakeelmohamed commented on .stylelintrc:
```diff
@@ -47,13 +46,10 @@
     "media-query-list-comma-newline-before": "never-multi-line",
     "media-query-list-comma-space-after": "always-single-line",
     "media-query-list-comma-space-before": "never",
-    "media-query-parentheses-space-inside": "always",
```
I'd prefer to revert this change just for my preference of style

shakeelmohamed commented on css/styles.css:
```diff
@@ -113,3 +152,77 @@ header, footer {
+
+.clear-both {
+    clear: both;
+    height: 0px;
```
Can we change this to just `0;`?

shakeelmohamed replied:
ping @YuriBrunetto

YuriBrunetto replied:
Sorry, I was out all day... Yeah, it could be without the `px`. I will do that now

shakeelmohamed replied:
No worries, just wanted to make sure you didn't forget about that comment

shakeelmohamed commented on test/index.js:
```diff
@@ -70,14 +70,14 @@ describe("Page Structure", function () {
         assert.equal(browser.query("header > figure > a").href, "https://zenplayer.audio/");
-        assert.ok(browser.query("header > figure > a > img.logo").src.indexOf("img/zen-audio-player-905.png") !== -1);
-        assert.equal(browser.query("header > figure > a > img.logo").alt, "Zen Audio Player logo");
+        // assert.ok(browser.query("header > figure > a > img.logo").src.indexOf("img/zen-audio-player-905.png") !== -1);
```
Not a fan of commenting out test, but I've created #164 and I can fix it myself no worries :smile:

YuriBrunetto replied:
ahh, okay!! thanks :heart:

shakeelmohamed commented on css/styles.css:
```diff
@@ -113,3 +152,77 @@ header, footer {
+
+    .input-search {
+        height: 50px;
+        font-size: 18px !important;
```
Do we really need `!important` here?

shakeelmohamed commented on css/styles.css:
```diff
@@ -113,3 +152,77 @@ header, footer {
+}
+
+.color-grey {
+    color: #666666 !important;
```
Do we really need !important here?

shakeelmohamed commented on .stylelintrc:
```diff
@@ -24,7 +24,6 @@
     "declaration-colon-newline-after": "always-multi-line",
     "declaration-colon-space-after": "always",
     "declaration-colon-space-before": "never",
-    "declaration-no-important": true,
```
I'd like to see if we can get away with reverting this

shakeelmohamed commented on .stylelintrc:
```diff
@@ -47,13 +46,10 @@
     "number-leading-zero": "always",
-    "number-max-precision": 2,
     "number-no-trailing-zeros": true,
-    "number-zero-length-no-unit": true,
```
We should be able to revert this. Is there a good reason for removing it?

agauniyal replied:
`number-max-precision` could be taken away but `number-zero-length-no-unit` is still a good rule that shouldn't be ignored.

YuriBrunetto replied:
It's because of the responsive, after a certain width some elements doesn't need `margin-top` anymore. Do you have some idea of what could we do?

shakeelmohamed replied:
@YuriBrunetto when does it make sense to use `0px` instead of just `0`?

agauniyal commented:
@YuriBrunetto Here is the list of rules stylelint checks for - https://github.com/stylelint/stylelint/blob/master/docs/user-guide/rules.md

YuriBrunetto commented:
@agauniyal i've never used stylelint before :ll but i'll read it 

and the way I write my css it's a bit different then the normal... it's like `.element { width:100px; height:30px; }` and I think this way keep the things more clean :smile:

shakeelmohamed commented:
LGTM - update your branch and I'll merge it in. Click the link in the box to see the directions on the command line
![capturfiles](https://cloud.githubusercontent.com/assets/1480063/13732482/0b88d868-e941-11e5-949b-99a7d1539cda.png)

YuriBrunetto commented:
@shakeelmohamed and @agauniyal:

I've removed the `!important` and `0px` and added `"number-zero-length-no-unit": true,` and `"declaration-no-important": true,` to `.stylelintrc` :+1: :smile:

shakeelmohamed commented:
Thanks for the effort @YuriBrunetto, welcome to the org!

YuriBrunetto commented:
Thanks, @shakeelmohamed! I'm very proud to be here :smile_cat:
